python_modules/libraries/dagster-k8s/dagster_k8s/ops/k8s_job_op.py

```diff
@@ -377,33 +378,38 @@ def execute_k8s_job(
                 start_time=start_time,  # pyright: ignore[reportArgumentType]
             )
 
-            log_stream = watch.stream(
-                api_client.core_api.read_namespaced_pod_log,
-                name=pod_to_watch,
-                namespace=namespace,
-                container=container_name,
-            )
-
             while True:
-                if timeout and time.time() - start_time > timeout:
-                    watch.stop()
-                    raise Exception("Timed out waiting for pod to finish")
                 try:
-                    log_entry = k8s_api_retry(
-                        lambda: next(log_stream),
-                        max_retries=int(
-                            os.getenv("DAGSTER_EXECUTE_K8S_JOB_STREAM_LOGS_RETRIES", "3")
-                        ),
-                        timeout=int(
-                            os.getenv(
-                                "DAGSTER_EXECUTE_K8S_JOB_STREAM_LOGS_WAIT_BETWEEN_ATTEMPTS", "5"
-                            )
-                        ),
+                    log_stream = watch.stream(
+                        api_client.core_api.read_namespaced_pod_log,
+                        name=pod_to_watch,
+                        namespace=namespace,
+                        container=container_name,
                     )
-                    print(log_entry)  # noqa: T201
+                    while True:
+                        if timeout and time.time() - start_time > timeout:
+                            watch.stop()
+                            raise Exception("Timed out waiting for pod to finish")
+
+                        log_entry = k8s_api_retry(
+                            lambda: next(log_stream),
+                            max_retries=int(
+                                os.getenv("DAGSTER_EXECUTE_K8S_JOB_STREAM_LOGS_RETRIES", "3")
+                            ),
+                            timeout=int(
+                                os.getenv(
+                                    "DAGSTER_EXECUTE_K8S_JOB_STREAM_LOGS_WAIT_BETWEEN_ATTEMPTS", "5"
+                                )
+                            ),
+                        )
+                        print(log_entry)  # noqa: T201
                 except StopIteration:
                     break
-        else:
+                except ProtocolError as e:
+                    context.log.warning(
+                        f"urllib3.exceptions.ProtocolError. Pausing and will reconnect. {e}"
```

Comment:
I do have one question actually - will this cause the logs to start over from the beginning if some have already been output before this error has been raised?

You could also imagine it saying that on any error reading the logs, including this one, it logs a warning with the stack trace explaining what the failure was, but doesn't fail the whole op (or try to start over) and just continues on, waiting for the pod to finsih

Reply:
It's hard to tell from the docs, but I just checked our logs and I can confirm that it creates duplicates logs. I can't find an easy way to fix that (we could maybe use `since_seconds`, but according to the docs it the time relative to "now", so we'd need to be careful there). Still, I'd rather have duplicated logs than a failed op :)

As for your suggestion - are you suggesting that if it encounters this (or other) failures, it'll just give a warning saying to it failed getting the logs (so logs could be partial), and then wait for the job to complete? I guess it's an option too. If that's the preferred behavior, I can modify the code, just let me know what you prefer.

(also - woohoo! I just confirmed that some jobs that this fix prevented some of our jobs from failing)